### PR TITLE
NFS and SMB mounts refactoring

### PR DESCRIPTION
**Pull request by lmakarov**
- NFS and SMB mounts configuration refactoring
- Fixed NFS export access with Docker for Mac
- Configure `nfs.server.mount.require_resv_port = 0` automatically
- Matched NFS options in `volumes-nfs.yml` with the NFS options in `docker_machine_mount_nfs`

### CODE REVIEW

**Comment by lmakarov** [comment]
Tested on Mac + VirtualBox and Docker for Mac.
Need to test with Windows + VirtualBox.

@achekulaev please start reviewing/testing when you have time.

**Comment by lmakarov** [comment]
Decided to do a fresh comparison of mount options on Docker Desktop (Mac):

docksal misc comparison - google sheets 2018-12-01 01-28-52

Docker for Mac + NFS delivers the same performance/overhead ratio as with VirtualBox + NFS - about 10% overhead on top of natives file system performance. The only exception is the first ("cold"?) run.

`osxfs:cached` is about 2x slower than native fs (40-50% slower than NFS).
`osxfs` (added for completeness) is about 3x slower than native fs.

**Comment by achekulaev** [comment]
It's unfinished right @lmakarov ?

**Comment by lmakarov** [comment]
@achekulaev the PR is ready, but I have not tested this on Windows, since I don't have one with VirtualBox working ATM. My VMWare Fusion Windows keeps crashing with VirtualBox...

**Comment by lmakarov** [comment]
Rebased onto develop.

File: bin/fin [hunk 1]
```diff
@@ -2292,68 +2272,81 @@ docker_machine_mount_nfs ()
...
+
+# Mount folder inside docker-machine with NFS
+# Unnamed params - shares in format "$LOCAL_FOLDER:$MOUNT_POINT_NAME"
+docker_machine_mount_nfs ()
+{
+	local machine_name="$DOCKER_MACHINE_NAME"
+	local nfs_ip="$DOCKSAL_HOST_IP"
+	eval $(parse_params "$@")
+
+	machine_ip=$(docker-machine ip "$machine_name")
```
Comment (achekulaev) [inline bin/fin]: Unused var

**Pull request merged**